deso/block.go

```diff
@@ -944,6 +956,161 @@ func (node *Node) GetValidatorPKIDFromSubAccountIdentifier(subAccount *types.Sub
 	return lib.NewPKID(validatorPKIDBytes), nil
 }
 
+func (node *Node) getStakeOps(txn *lib.MsgDeSoTxn, utxoOps []*lib.UtxoOperation, numOps int) []*types.Operation {
+	if txn.TxnMeta.GetTxnType() != lib.TxnTypeStake {
+		return nil
+	}
+
+	var operations []*types.Operation
+	for _, utxoOp := range utxoOps {
+		if utxoOp.Type == lib.OperationTypeStake {
+			prevValidatorEntry := utxoOp.PrevValidatorEntry
+			if prevValidatorEntry == nil {
+				// TODO: This is a bad error...
+				glog.Error("getStakeOps: prevValidatorEntry was nil")
+				continue
+			}
+			realTxMeta := txn.TxnMeta.(*lib.StakeMetadata)
+			if realTxMeta == nil {
+				glog.Error("getStakeOps: realTxMeta was nil")
+				continue
+			}
+			stakeAmountNanos := realTxMeta.StakeAmountNanos
+			if !stakeAmountNanos.IsUint64() {
+				glog.Error("getStakeOps: stakeAmountNanos was not a uint64")
+				continue
+			}
+			stakeAmountNanosUint64 := stakeAmountNanos.Uint64()
+
+			operations = append(operations, &types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: int64(numOps),
+				},
+				Account: &types.AccountIdentifier{
+					Address:    lib.Base58CheckEncode(prevValidatorEntry.ValidatorPKID.ToBytes(), false, node.Params),
+					SubAccount: &types.SubAccountIdentifier{Address: ValidatorEntry},
+				},
+				Amount: &types.Amount{
+					Value:    strconv.FormatUint(stakeAmountNanosUint64, 10),
+					Currency: &Currency,
+				},
+				Status: &SuccessStatus,
+				Type:   OutputOpType,
+			})
+			numOps++
+		}
+	}
+	return operations
+}
+
+func (node *Node) getUnstakeOps(txn *lib.MsgDeSoTxn, utxoOps []*lib.UtxoOperation, numOps int) []*types.Operation {
+	if txn.TxnMeta.GetTxnType() != lib.TxnTypeUnstake {
+		return nil
+	}
+
+	var operations []*types.Operation
+	for _, utxoOp := range utxoOps {
+		if utxoOp.Type == lib.OperationTypeUnstake {
+			prevValidatorEntry := utxoOp.PrevValidatorEntry
+			if prevValidatorEntry == nil {
+				glog.Error("getUnstakeOps: prevValidatorEntry was nil")
+				continue
+			}
+			prevStakeEntries := utxoOp.PrevStakeEntries
+			if len(prevStakeEntries) != 1 {
+				glog.Error("getUnstakeOps: prevStakeEntries was not of length 1")
+				continue
+			}
+			prevStakeEntry := prevStakeEntries[0]
+			realTxMeta := txn.TxnMeta.(*lib.UnstakeMetadata)
+			if realTxMeta == nil {
+				glog.Error("getUnstakeOps: realTxMeta was nil")
+				continue
+			}
+			unstakeAmountNanos := realTxMeta.UnstakeAmountNanos
+			if !unstakeAmountNanos.IsUint64() {
+				glog.Error("getUnstakeOps: unstakeAmountNanos was not a uint64")
+				continue
+			}
+			unstakeAmountNanosUint64 := unstakeAmountNanos.Uint64()
+			// First use an "input" from the ValidatorEntry
+			operations = append(operations, &types.Operation{
+				OperationIdentifier: &types.OperationIdentifier{
+					Index: int64(numOps),
+				},
+				Account: &types.AccountIdentifier{
```

Comment:
I kinda widh you had a function like getAccountIdentifierForX() like you have for other things, but no need if it works.

Reply:
done!